modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/http/netty4/Netty4ChunkedEncodingIT.java

```diff
@@ -0,0 +1,151 @@
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.bytes.ReleasableBytesReference;
+import org.elasticsearch.common.collect.Iterators;
+import org.elasticsearch.common.io.Streams;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.recycler.Recycler;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.IndexScopedSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.settings.SettingsFilter;
+import org.elasticsearch.common.util.CollectionUtils;
+import org.elasticsearch.plugins.ActionPlugin;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.ChunkedRestResponseBody;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestHandler;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestResponse;
+import org.elasticsearch.rest.RestStatus;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Supplier;
+
+import static org.elasticsearch.rest.RestRequest.Method.GET;
+import static org.elasticsearch.rest.RestResponse.TEXT_CONTENT_TYPE;
+import static org.hamcrest.Matchers.containsString;
+
+public class Netty4ChunkedEncodingIT extends ESNetty4IntegTestCase {
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return CollectionUtils.concatLists(List.of(YieldsChunksPlugin.class), super.nodePlugins());
+    }
+
+    @Override
+    protected boolean addMockHttpTransport() {
+        return false; // enable http
+    }
+
+    private static final String expectedBody = """
+        chunk-0
+        chunk-1
+        chunk-2
+        """;
+
+    public void testBasic() throws IOException {
+        final var response = getRestClient().performRequest(new Request("GET", YieldsChunksPlugin.ROUTE));
+        assertEquals(200, response.getStatusLine().getStatusCode());
+        assertThat(response.getEntity().getContentType().toString(), containsString(TEXT_CONTENT_TYPE));
+        assertTrue(response.getEntity().isChunked());
+        final String body;
+        try (var reader = new InputStreamReader(response.getEntity().getContent(), StandardCharsets.UTF_8)) {
+            body = Streams.copyToString(reader);
+        }
+        assertEquals(expectedBody, body);
+    }
+
+    public static class YieldsChunksPlugin extends Plugin implements ActionPlugin {
+        static final String ROUTE = "/_test/yields_chunks";
+
+        public static class Request extends ActionRequest {
+            @Override
+            public ActionRequestValidationException validate() {
+                return null;
+            }
+        }
```

Comment:
This does not seem to be used anywhere.

Reply:
Ah yes, a leftover from an earlier test.